userspace/libscap/examples/01-open/scap_open.c

```diff
@@ -20,42 +20,328 @@ limitations under the License.
+/// TODO: we need to move this validation outside this example
+
+void validate_syscalls()
+{
+	enum ppm_syscall_code ppm_syscall_code = 0;
+	bool success = true;
+	/* For every syscall of the system. */
+	for(int syscall_id = 0; syscall_id < SYSCALL_TABLE_SIZE; syscall_id++)
+	{
+
+		ppm_syscall_code = g_syscall_code_routing_table[syscall_id];
+		/* If the syscall has `UF_NEVER_DROP` flag we must have its name inside the
+		 * `g_syscall_info_table`.
+		 */
+		if(g_syscall_table[syscall_id].flags & UF_NEVER_DROP && !g_syscall_info_table[ppm_syscall_code].name)
+		{
+			printf("ERROR: the syscall with real id `%d` has a `UF_NEVER_DROP` syscall in `g_syscall_table` but not a name in the `g_syscall_info_table`.\n", syscall_id);
+			success = false;
+			continue;
+		}
+
+		if(g_syscall_table[syscall_id].enter_event_type == PPME_GENERIC_E)
+		{
+			continue;
+		}
+
+		/* This is an error since it means that a syscall we want to trace is not tracked in our `g_syscall_info_table`.
+		 * We have `EC_UNKNOWN` when we don't have an entry in the `g_syscall_info_table`.
+		 */
+		if(g_syscall_info_table[ppm_syscall_code].category == EC_UNKNOWN)
+		{
+			printf("ERROR: the syscall with ppm code '%d' has an event associated but it is unknown in our `g_syscall_info_table`.\n", ppm_syscall_code);
+			success = false;
+			continue;
+		}
+	}
+
+	if(success)
+	{
+		printf("\n[SUCCESS] Our table are consistent!\n");
+	}
+	else
+	{
+		printf("\n[FAIL] Our table are not consistent!\n");
+	}
+}
+
+/*=============================== PRINT SUPPORTED SYSCALLS ===========================*/
+
+/*=============================== SYSCALLS/TRACEPOINTS ===========================*/
+
+void enable_single_tp(int tp)
+{
+	if(tp == -1)
+	{
+		/* In this case we won't have any tracepoint enabled. */
+		tp_is_set = true;
+		return;
+	}
+
+	if(tp < 0 || tp >= TP_VAL_MAX)
+	{
+		fprintf(stderr, "Unexistent tp code: %d. Wrong parameter?\n", tp);
+		print_supported_tracepoints();
+		exit(EXIT_FAILURE);
+	}
+	oargs.tp_of_interest.tp[tp] = true;
+	tp_is_set = true;
+}
+
+void enable_single_ppm_sc(int ppm_sc_code)
+{
+	if(ppm_sc_code == -1)
+	{
+		/* In this case we won't have any syscall enabled. */
+		ppm_sc_is_set = true;
+		return;
+	}
+
+	if(ppm_sc_code < 0 || ppm_sc_code >= PPM_SC_MAX)
+	{
+		fprintf(stderr, "Unexistent ppm_sc code: %d. Wrong parameter?\n", ppm_sc_code);
+		print_supported_syscalls();
+		exit(EXIT_FAILURE);
+	}
+	oargs.ppm_sc_of_interest.ppm_sc[ppm_sc_code] = true;
+	ppm_sc_is_set = true;
+}
+
+void check_enabled_syscalls()
+{
+	printf("---------------------- INTERESTING SYSCALLS ----------------------\n");
+	if(ppm_sc_is_set)
+	{
+		printf("* Syscalls enabled:\n");
+		for(int j = 0; j < PPM_SC_MAX; j++)
+		{
+			if(oargs.ppm_sc_of_interest.ppm_sc[j])
+			{
+				printf("- %s\n", g_syscall_info_table[j].name);
+			}
+		}
+	}
+	else
+	{
+		printf("* All syscalls are enabled!\n");
+		for(int j = 0; j < PPM_SC_MAX; j++)
+		{
+			oargs.ppm_sc_of_interest.ppm_sc[j] = true;
```

Comment:
I feel like this is not something we want to do in a `check` function, the rest of the call seems to be strictly for printing information but then there is this weird side effect were not calling the check function will not actually enable all syscalls. It's not major, but it does look like it might make it harder to trace the code on future changes.